packages/dds/tree/src/test/domains/json/twitter.ts

```diff
@@ -0,0 +1,631 @@
+* generated json may be closer to the desired byte size but there is a risk of exceeding the inputted byte limit
+* @returns TwitterJson
+*/
+export function generateTwitterJsonByByteSize(sizeInBytes: number, allowOversize: boolean, seed = 1) {
+    const random = makeRandom(seed);
+    const textFieldMarkovChain = new SpaceEfficientMarkovChain(random, getTwitterJsonTextFieldMarkovChain());
+    const userDescFieldMarkovChain = new SpaceEfficientMarkovChain(random, getTwitterJsonUserDescFieldMarkovChain());
+    const basicJapaneseAlphabetString = getBasicJapaneseAlphabetString();
+    const twitterJson: TwitterJson = {
+        statuses: [],
+        search_metadata: {
+            completed_in: 0.087,
+            max_id: 505874924095815700,
+            max_id_str: "505874924095815681",
+            next_results: "?max_id=505874847260352512&q=%E4%B8%80&count=100&include_entities=1",
+            query: "%E4%B8%80",
+            refresh_url: "?since_id=505874924095815681&q=%E4%B8%80&include_entities=1",
+            count: 100,
+            since_id: 0,
+            since_id_str: "0",
+        },
+    };
+
+    let currentJsonSizeInBytes = getSizeInBytes(twitterJson);
+    while (currentJsonSizeInBytes < sizeInBytes) {
+        const twitterStatus = generateTwitterStatus(
+            "standard", random, textFieldMarkovChain, userDescFieldMarkovChain, basicJapaneseAlphabetString,
+        );
+        const nextStatusSizeInBytes = getSizeInBytes(twitterStatus);
+        if (!allowOversize && currentJsonSizeInBytes + nextStatusSizeInBytes > sizeInBytes) {
+            break;
+        }
+        twitterJson.statuses.push(twitterStatus);
+        currentJsonSizeInBytes += nextStatusSizeInBytes;
+    }
+
+    return twitterJson;
+}
+
+/**
+* Generates a TwitterJson object containing exactly the number specified statuses.
+* @param numStatuses - number of statuses to include in the generated TwitterJson
+* @param includeUnicode - true to include unicode in any strings within the json
+* @returns TwitterJson
+*/
+export function generateTwitterJsonByNumStatuses(numStatuses: number, seed = 1) {
+    const random = makeRandom(seed);
+    const textFieldMarkovChain = new SpaceEfficientMarkovChain(random, getTwitterJsonTextFieldMarkovChain());
+    const userDescFieldMarkovChain = new SpaceEfficientMarkovChain(random, getTwitterJsonUserDescFieldMarkovChain());
+    const basicJapaneseAlphabetString = getBasicJapaneseAlphabetString();
+    const twitterJson: TwitterJson = {
+        statuses: [],
+        search_metadata: {
+            completed_in: 0.087,
+            max_id: 505874924095815700,
+            max_id_str: "505874924095815681",
+            next_results: "?max_id=505874847260352512&q=%E4%B8%80&count=100&include_entities=1",
+            query: "%E4%B8%80",
+            refresh_url: "?since_id=505874924095815681&q=%E4%B8%80&include_entities=1",
+            count: 100,
```

Comment:
Nit: I suspect 'count' is supposed to be equal to num statuses.

Reply:
Done